### ISSUE TITLE
Warnings about compiler-libs when using ppx_blob

### ISSUE THREAD

**Issue by hcarty** [johnwhitington/ppx_blob#1]
I'm not sure if this is an issue with ppx_blob or elsewhere.  I get these warnings when compiling anything using ppx_blob using OCaml 4.02.1/4.02.2 and ppx_blob 0.1.  Everything else is the latest from opam.

`$ cat test.ml`

``` ocaml
let () =
  let s = [%blob "test.ml"] in
  print_endline s
```

```
$ ocamlbuild -use-ocamlfind -package ppx_blob test.native
+ ocamlfind ocamlc -c -package ppx_blob -o test.cmo test.ml
findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/hcarty/.opam/4.02.2/lib/ocaml, /home/hcarty/.opam/4.02.2/lib/ocaml/compiler-libs
+ ocamlfind ocamlopt -c -package ppx_blob -o test.cmx test.ml
findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/hcarty/.opam/4.02.2/lib/ocaml, /home/hcarty/.opam/4.02.2/lib/ocaml/compiler-libs
+ ocamlfind ocamlopt -linkpkg -package ppx_blob test.cmx -o test.native
findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/hcarty/.opam/4.02.2/lib/ocaml, /home/hcarty/.opam/4.02.2/lib/ocaml/compiler-libs
Finished, 4 targets (0 cached) in 00:00:00.
```

**Comment by johnwhitington** [comment]
I see this too.

Can you try switching to your system compiler, if it's recent enough, to see if this is OPAM related or also happens with vanilla OCaml installs?

**Comment by hcarty** [comment]
I tested with an installation from avsm's ppa packages on Ubuntu 12.04 (4.02.1-1ppa3~precise) and get the same warnings.

**Comment by hcarty** [comment]
There is a bug on OCaml's mantis for this: http://caml.inria.fr/mantis/view.php?id=6754

Moving the `lib/ocaml/topdirs.cmi` file out of the way (for example to `lib/ocaml/topdirs.cmi.bak`) silences the warnings.  I had to add `#directory "+compiler-libs";;` to the top of my `.ocamlinit` file.

**Comment by hcarty** [comment]
Is it enough to remove `requires="ppx_tools"` from the META file?

Edit: Sorry for the noise.  I see that you've already made that change.  Could you make another release?

**Comment by johnwhitington** [comment]
I did remove ppx_tools, but ISTR some sort of failure, so I didn't cut a release. I'll have another look at it this weekend for you.

**Comment by johnwhitington** [comment]
Sorry for the delay. This is the issue blocking  a release of with ppx_tools dependency removed:

https://github.com/ocaml/opam-repository/issues/4043

I have no idea what is needed to fix it.

**Comment by aantron** [comment]
FYI the warning itself can be worked around as here: https://github.com/aantron/bisect_ppx/commit/19babfdfc51dbdd1bbb70752c819de61839cdecf

**Comment by johnwhitington** [comment]
@aantron: Do you have any idea about https://github.com/ocaml/opam-repository/issues/4043 ?

**Comment by aantron** [comment]
Yes, the `META` file in the released archive `https://github.com/johnwhitington/ppx_blob/archive/v0.1.tar.gz`says that package `ppx_blob` depends on `ppx_tools`, but the `opam` file does not ask it to be installed. I think you removed the dependency on ppx_tools _after_ releasing 0.1. The old `opam` file should be amended with the dependency.

https://github.com/ocaml/opam-repository/issues/4043 won't be an issue in the PR I'm about to submit – I've added an `opam` file that depends on `ppx_tools`.

This issue (#1) was fixed by your commit https://github.com/johnwhitington/ppx_blob/commit/1dbbe4496b3c5cdc143f61ef0b8dd2afbe6609f4 and can be closed. Package `ppx_blob` does not require `ppx_tools` at link time (only at build time). It is correct to remove the Findlib dependency as in https://github.com/johnwhitington/ppx_blob/commit/1dbbe4496b3c5cdc143f61ef0b8dd2afbe6609f4, and that has the side effect of getting rid of this warning for users of `ppx_blob`.
